diff --git a/dist/index.js b/dist/index.js
index f2c306c87f6110dc90dd8cbafd0c9e75f59c81c4..ef0aaa1ae30b047bafd49fa3468e22620ddaa73b 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3270,7 +3270,18 @@ var devServerEndPointsPlugin = ({
           return;
         }
         if (req.url === "/graphql") {
-          const { query, variables } = req.body;
+          const { query, variables } = req.body ?? {};
+          if (typeof query !== "string") {
+            res.statusCode = 400;
+            res.end(
+              JSON.stringify({
+                errors: [
+                  { message: "No GraphQL query present in request body" }
+                ]
+              })
+            );
+            return;
+          }
           let result;
           await databaseLock(async () => {
             result = await gqlResolve({
